libs/dyn/dynvar/ref.go

```diff
@@ -22,6 +23,8 @@ type ref struct {
 	matches [][]string
 }
 
+var invalidSeq = []string{"-_", "_-"}
+
```

Comment:
Why not let the regex deal with this?

We currently restrict to the following:
1. The first character of a field must be `a-z`
2. The following chunks must be `a-z0-9`
3. The chunks may be separated by `_*` or `-*`
4. It can have a trailer of one or more indices `(\[[0-9]+\])*`

I think the simplified base regex for this looks like:
```
([a-zA-Z]+((_*|-*)[a-zA-Z0-9]+)*(\[[0-9]+\])*)
```

And then we can have one or more of these, separated by a `.`:
```
(${base})(\.${base})*
```

What is the nuance behind not allowing `_-` btw?

Reply:
It's a bit weird pattern but actually valid for both YAML and Terraform, so let's keep it